editor/sidebar/block-inspector/index.js

```diff
@@ -18,36 +19,76 @@ import './style.scss';
 import { deselectBlock } from '../../actions';
 import { getSelectedBlock } from '../../selectors';
 
-const BlockInspector = ( { selectedBlock, ...props } ) => {
-	if ( ! selectedBlock ) {
-		return null;
+class BlockInspector extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.forceUpdateIfSelected = this.forceUpdateIfSelected.bind( this );
+	}
+
+	componentWillMount() {
+		const { manager: rsfManager } = this.context;
+		rsfManager.onComponentsChange( 'Inspector.Controls', this.forceUpdateIfSelected );
+	}
+
+	componentWillUnmount() {
+		const { manager: rsfManager } = this.context;
+		rsfManager.removeOnComponentsChange( 'Inspector.Controls', this.forceUpdateIfSelected );
+	}
+
+	forceUpdateIfSelected() {
+		// We must bind to the React-Slot-Fill Provider manager to re-render if
+		// slots change after the initial render. This can occur when focus
+		// changes between blocks, where control fills may only update after
+		// change in selected block (and subsequent render) already occurred.
+		if ( this.props.selectedBlock ) {
+			this.forceUpdate();
```

Comment:
>because by definition a React component should rerender if its props or state updates.

This is true in normal cases, I agree. `context` falls outside of this, however, and doesn't follow the same criteria. One can't set like `setState`, nor re-render with new `props`, so manually binding to the events is the next best option. We must force the render one way or another; `setState` is an option, but without refactoring falsely implies there's state of the component when in fact we just care to adopt its re-rendering implication. `forceUpdate` does the same without the additional connotation of state existing.

>What state value are we duplicating here?

If we consider the source of truth to be React-Slot-Fill's internal manager, we needn't create a copy of this data (nor derived data) and instead call to it directly as part of the render.

Let me tinker with this a bit longer and see if I can't find another approach.